src/main/java/slimeknights/tconstruct/tools/tools/Scythe.java

```diff
@@ -1,158 +1,306 @@
-        for(int zp = start.getZ() - d; zp != start.getZ() + d; zp++) {
-          if(MathHelper.abs_int(xp - start.getX()) + MathHelper.abs_int(yp - start.getY()) + MathHelper
-              .abs_int(zp - start.getZ()) >= distance) {
-            continue;
+  public ActionResult<ItemStack> onItemRightClick(@Nonnull ItemStack stack, World world, EntityPlayer player, EnumHand hand) {
+    if(ToolHelper.isBroken(stack)) {
+      return ActionResult.newResult(EnumActionResult.FAIL, stack);
+    }
+
+    RayTraceResult trace = this.rayTrace(world, player, false);
+    if(trace == null || trace.typeOfHit != RayTraceResult.Type.BLOCK) {
+      return ActionResult.newResult(EnumActionResult.PASS, stack);
+    }
+
+    int fortune = EnchantmentHelper.getEnchantmentLevel(Enchantments.FORTUNE, stack);
+
+    // if we cannot harvest the center block then back out
+    BlockPos origin = trace.getBlockPos().offset(trace.sideHit);
+    if(!harvestBlock(stack, world, player, origin, fortune)) {
+      return ActionResult.newResult(EnumActionResult.PASS, stack);
+    }
+
+    // otherwise if we succeed harvest the rest
+    for(BlockPos pos : ((IAoeTool) this).getAOEBlocks(stack, player.worldObj, player, origin)) {
+      harvestBlock(stack, world, player, pos, fortune);
+    }
+
+    player.worldObj.playSound(null, player.posX, player.posY, player.posZ, SoundEvents.ENTITY_PLAYER_ATTACK_SWEEP, player.getSoundCategory(), 1.0F, 1.0F);
+    player.spawnSweepParticles();
+
+    return ActionResult.newResult(player.worldObj.isRemote ? EnumActionResult.PASS : EnumActionResult.SUCCESS, stack);
+  }
+
+  /**
+   * Returns true if the item can be used on the given entity, e.g. shears on sheep.
+   */
+  @Override
+  public boolean itemInteractionForEntity(ItemStack stack, EntityPlayer player, EntityLivingBase target, EnumHand hand) {
+    // only run AOE on shearable entities
+    if(!(target instanceof IShearable)) {
+      return false;
+    }
+
+    // increase the size based on the AOE stuffs
+    TinkerToolEvent.ExtraBlockBreak event = TinkerToolEvent.ExtraBlockBreak.fireEvent(stack, player, player.worldObj.getBlockState(target.getPosition()), 3, 3, 3, -1);
+    if(event.isCanceled()) {
+      return false;
+    }
+
+    int distance = event.distance;
+    boolean shorn = false;
+
+    int fortune = EnchantmentHelper.getEnchantmentLevel(Enchantments.FORTUNE, stack);
+    for(Entity entity : getAoeEntities(player, target, event)) {
+      if(distance < 0 || entity.getDistanceToEntity(target) <= distance) {
+        shorn |= shearEntity(stack, player.worldObj, player, entity, distance, fortune);
+      }
+    }
+
+    if(shorn) {
+      player.worldObj.playSound(null, player.posX, player.posY, player.posZ, SoundEvents.ENTITY_PLAYER_ATTACK_SWEEP, player.getSoundCategory(), 1.0F, 1.0F);
+      player.spawnSweepParticles();
+    }
+
+    return shorn;
+  }
+
+  public boolean harvestBlock(ItemStack stack, World world, EntityPlayer player, BlockPos pos, int fortune) {
```

Comment:
Best rename to something more appropriate so it's not confused with the regular leftclick blockbreaking (whose functions are all called harvest)